crates/maple_cli/src/cmd/filter.rs

```diff
@@ -70,3 +70,409 @@ pub fn blines(
         winwidth,
     )
 }
+
+/// Return the ranked results after applying fuzzy filter given the query string and a list of candidates.
+pub fn dyn_fuzzy_filter_and_rank<I: Iterator<Item = String>>(
```

Comment:
I think it's worthy of moving all the dyn_* part into a new module under `filter`, putting the const items on the top, moving the trait/macro outside the functions.